src/process/namespaces.rs

```diff
@@ -15,13 +10,33 @@ impl Process {
     /// Doc reference: https://man7.org/linux/man-pages/man7/namespaces.7.html
     /// The namespace type is the key for the HashMap, i.e 'net', 'user', etc.
     pub fn namespaces(&self) -> ProcResult<HashMap<OsString, Namespace>> {
-        let ns = self.root.join("ns");
         let mut namespaces = HashMap::new();
-        for entry in fs::read_dir(ns)? {
-            let entry = entry?;
-            let path = entry.path();
-            let ns_type = entry.file_name();
-            let stat = rustix::fs::statat(&rustix::fs::cwd(), &path, AtFlags::empty())
+        let dir_ns = wrap_io_error!(
+            self.root.join("ns"),
+            rustix::fs::openat(
+                &self.fd,
+                "ns",
+                OFlags::RDONLY | OFlags::DIRECTORY | OFlags::CLOEXEC,
+                Mode::empty()
+            )
+        )?;
+        let dir = wrap_io_error!(self.root.join("ns"), rustix::fs::Dir::read_from(&dir_ns))?;
+        for entry in dir {
+            let entry = entry.map_err(|_| build_internal_error!(format!("Unable to get ns dir entry")))?;
+            match entry.file_name().to_string_lossy().as_ref() {
+                "." | ".." => continue,
+                _ => {}
+            };
+
+            let path = self
+                .root
+                .join("ns")
+                .join(entry.file_name().to_string_lossy().to_string());
+            let ns_type = OsString::from(entry.file_name().to_string_lossy().to_string());
+            let dir_ns = dir_ns
+                .try_clone()
+                .map_err(|_| build_internal_error!(format!("Unable to clone ns dir")))?;
+            let stat = rustix::fs::statat(dir_ns, entry.file_name(), AtFlags::empty())
```

Comment:
I don't think this clone is necessary

```suggestion
            let dir_ns = dir_ns
                .try_clone()
                .map_err(|_| build_internal_error!(format!("Unable to clone ns dir")))?;
            let stat = rustix::fs::statat(dir_ns, entry.file_name(), AtFlags::empty())
```
```suggestion
            let stat = rustix::fs::statat(&dir_ns, entry.file_name(), AtFlags::empty())
```

Reply:
Done